--- ruby-1.9.3-p547/missing/setproctitle.c	2013-02-05 21:13:40.000000000 -0800
+++ ruby-1.9.3-p547-patched/missing/setproctitle.c	2014-10-08 09:58:12.000000000 -0700
@@ -70,6 +70,7 @@
 static char *argv_start = NULL;
 static size_t argv_env_len = 0;
 static size_t argv_len = 0;
+static char **argv1_addr = NULL;
 #endif
 
 #endif /* HAVE_SETPROCTITLE */
@@ -115,7 +116,9 @@
 			lastenvp = envp[i] + strlen(envp[i]);
 	}
 
-	argv[1] = NULL;
+	/* We keep argv[1], argv[2], etc. at this moment,
+		 because the ps command of AIX refers to them. */
+	argv1_addr = &argv[1];
 	argv_start = argv[0];
 	argv_len = lastargv - argv[0];
 	argv_env_len = lastenvp - argv[0];
@@ -158,6 +161,8 @@
 	argvlen = len > argv_len ? argv_env_len : argv_len;
 	for(; len < argvlen; len++)
 		argv_start[len] = SPT_PADCHAR;
+	/* argv[1], argv[2], etc. are no longer valid. */
+	*argv1_addr = NULL;
 #endif
 
 #endif /* SPT_NONE */
