From 7be64d42e09b75cf19f541e15890e769a8a561ee Mon Sep 17 00:00:00 2001
From: "Sean P. McDonald" <sean.mcdonald@puppet.com>
Date: Fri, 29 Jul 2016 14:37:38 -0700
Subject: [PATCH] update rbinstall for windows

---
 tool/rbinstall.rb | 45 +++++++++++++++++++++++----------------------
 1 file changed, 23 insertions(+), 22 deletions(-)

diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index b3dad0e..7d4124d 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -482,29 +482,30 @@ install?(:local, :comm, :bin, :'bin-comm') do
     shebang.sub!(/\r$/, '')
     body.gsub!(/\r$/, '')
 
-    cmd << ".#{$cmdtype}" if $cmdtype
+    #cmd << ".#{$cmdtype}" if $cmdtype
     open_for_install(cmd, $script_mode) do
-      case $cmdtype
-      when "exe"
-        stub + shebang + body
-      when "bat"
-        [<<-"EOH".gsub(/^\s+/, ''), shebang, body, "__END__\n:endofruby\n"].join.gsub(/$/, "\r")
-          @echo off
-          @if not "%~d0" == "~d0" goto WinNT
-          #{ruby_bin} -x "#{cmd}" %1 %2 %3 %4 %5 %6 %7 %8 %9
-          @goto endofruby
-          :WinNT
-          "%~dp0#{ruby_install_name}" -x "%~f0" %*
-          @goto endofruby
-        EOH
-      when "cmd"
-        <<"/EOH" << shebang << body
-@"%~dp0#{ruby_install_name}" -x "%~f0" %*
-@exit /b %ERRORLEVEL%
-/EOH
-      else
-        shebang + body
-      end
+#       case $cmdtype
+#       when "exe"
+#         stub + shebang + body
+#       when "bat"
+#         [<<-"EOH".gsub(/^\s+/, ''), shebang, body, "__END__\n:endofruby\n"].join.gsub(/$/, "\r")
+#           @echo off
+#           @if not "%~d0" == "~d0" goto WinNT
+#           #{ruby_bin} -x "#{cmd}" %1 %2 %3 %4 %5 %6 %7 %8 %9
+#           @goto endofruby
+#           :WinNT
+#           "%~dp0#{ruby_install_name}" -x "%~f0" %*
+#           @goto endofruby
+#         EOH
+#       when "cmd"
+#         <<"/EOH" << shebang << body
+# @"%~dp0#{ruby_install_name}" -x "%~f0" %*
+# @exit /b %ERRORLEVEL%
+# /EOH
+#       else
+#         shebang + body
+#       end
+      body
     end
   end
 end
-- 
2.7.4 (Apple Git-66)

