From 8f6913338eaae4f65a4b44cdc7a36a9be0337608 Mon Sep 17 00:00:00 2001
From: "Sean P. McDonald" <sean.mcdonald@puppetlabs.com>
Date: Thu, 30 Nov 2017 16:24:28 -0800
Subject: [PATCH] (maint) Remove O_CLOEXEC check for AIX builds

---
 io.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/io.c b/io.c
index ee3ea3e68a..eb21aa25b1 100644
--- a/io.c
+++ b/io.c
@@ -257,10 +257,7 @@ rb_cloexec_open(const char *pathname, int flags, mode_t mode)
     int ret;
     static int o_cloexec_state = -1; /* <0: unknown, 0: ignored, >0: working */
 
-#ifdef O_CLOEXEC
-    /* O_CLOEXEC is available since Linux 2.6.23.  Linux 2.6.18 silently ignore it. */
-    flags |= O_CLOEXEC;
-#elif defined O_NOINHERIT
+#ifdef defined O_NOINHERIT
     flags |= O_NOINHERIT;
 #endif
     ret = open(pathname, flags, mode);
-- 
2.14.2.windows.1

