From 3b30039b2ba4031bb064b1496e7c57fa6db9c36a Mon Sep 17 00:00:00 2001
From: "Sean P. McDonald" <sean.mcdonald@puppet.com>
Date: Thu, 28 Jul 2016 13:48:44 -0700
Subject: [PATCH] update rbinstall for windows

---
 tool/rbinstall.rb | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 4614c35..c2b395a 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -486,19 +486,20 @@ def CONFIG.[](name, mandatory = false)
     shebang.sub!(/\r$/, '')
     body.gsub!(/\r$/, '')
 
-    cmd << ".#{$cmdtype}" if $cmdtype
+#    cmd << ".#{$cmdtype}" if $cmdtype
     open_for_install(cmd, $script_mode) do
-      case $cmdtype
-      when "exe"
-        stub + shebang + body
-      when "cmd"
-        prebatch + <<"/EOH" << postbatch << shebang << body
-@"%~dp0#{ruby_install_name}" -x "%~f0" %*
-@exit /b %ERRORLEVEL%
-/EOH
-      else
-        shebang + body
-      end
+#       case $cmdtype
+#       when "exe"
+#         stub + shebang + body
+#       when "cmd"
+#         prebatch + <<"/EOH" << postbatch << shebang << body
+# @"%~dp0#{ruby_install_name}" -x "%~f0" %*
+# @exit /b %ERRORLEVEL%
+# /EOH
+#       else
+#         shebang + body
+#       end
+      body
     end
   end
 end
-- 
2.7.4 (Apple Git-66)

