From ad1177bc54cac2de0672d7188760ecf2d142e029 Mon Sep 17 00:00:00 2001
From: "Sean P. McDonald" <sean.mcdonald@puppet.com>
Date: Tue, 28 Jun 2016 13:24:22 -0700
Subject: [PATCH] Update missing script to return 0

---
 missing | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/missing b/missing
index f62bbae..4ef2bca 100755
--- a/missing
+++ b/missing
@@ -204,7 +204,7 @@ give_advice "$1" | sed -e '1s/^/WARNING: /' \
 
 # Propagate the correct exit status (expected to be 127 for a program
 # not found, 63 for a program that failed due to version mismatch).
-exit $st
+exit 0
 
 # Local variables:
 # eval: (add-hook 'write-file-hooks 'time-stamp)
-- 
2.7.4 (Apple Git-66)

