From 7654bd90461e27758316a447d13dd6fd9c0cb5d4 Mon Sep 17 00:00:00 2001
From: "Sean P. McDonald" <sean.mcdonald@puppet.com>
Date: Wed, 13 Jul 2016 15:10:30 -0700
Subject: [PATCH] win32ole-patch

---
 ext/win32ole/win32ole.c | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/ext/win32ole/win32ole.c b/ext/win32ole/win32ole.c
index 02ad82b..7a49ba0 100644
--- a/ext/win32ole/win32ole.c
+++ b/ext/win32ole/win32ole.c
@@ -864,7 +864,9 @@ ole_vstr2wc(VALUE vstr)
     /* do not type-conversion here to prevent from other arguments
      * changing (if exist) */
     Check_Type(vstr, T_STRING);
-
+    if (RSTRING_LEN(vstr) == 0) {
+        return NULL;
+    }
     enc = rb_enc_get(vstr);
 
     if (st_lookup(tbl, (VALUE)enc | FIXNUM_FLAG, &data)) {
@@ -1571,10 +1573,16 @@ ole_variant2val(VARIANT *pvar)
 
     case VT_BSTR:
     {
-        if(V_ISBYREF(pvar))
-            obj = ole_wc2vstr(*V_BSTRREF(pvar), FALSE);
-        else
-            obj = ole_wc2vstr(V_BSTR(pvar), FALSE);
+        if(V_ISBYREF(pvar)) {
+            obj = (SysStringLen(*V_BSTRREF(pvar)) == 0)
+                 ? rb_str_new2("")
+                 : ole_wc2vstr(*V_BSTRREF(pvar), FALSE);
+         }
+         else {
+             obj = (SysStringLen(V_BSTR(pvar)) == 0)
+                 ? rb_str_new2("")
+                 : ole_wc2vstr(V_BSTR(pvar), FALSE);
+        }
         break;
     }
 
-- 
2.7.4 (Apple Git-66)

