diff --git a/lib/puppet/util/run_mode.rb b/lib/puppet/util/run_mode.rb
index 9a736fb..4d70d26 100644
--- a/lib/puppet/util/run_mode.rb
+++ b/lib/puppet/util/run_mode.rb
@@ -55,11 +55,11 @@ module Puppet
 
     class UnixRunMode < RunMode
       def conf_dir
-        which_dir("/etc/puppet", "~/.puppet")
+        which_dir("/etc/puppetlabs/agent", "~/.puppet")
       end
 
       def var_dir
-        which_dir("/var/lib/puppet", "~/.puppet/var")
+        which_dir("/opt/puppetlabs/agent/cache", "~/.puppet/var")
       end
     end
 
